src/components/transport_manager/include/transport_manager/iap2_emulation/iap2_transport_adapter.h

```diff
@@ -105,6 +115,38 @@ class IAP2USBEmulationTransportAdapter : public TcpTransportAdapter {
    * @return Device type
    */
   DeviceType GetDeviceType() const OVERRIDE;
+
+  private:
+  /**
+   * @brief The IAPSignalHandlerDelegate class handles signals from the system
+   * to start transport switching flow
+   */
+  class IAPSignalHandlerDelegate : public threads::ThreadDelegate {
+    public:
+    /**
+     * @brief IAPSignalHandlerDelegate Constructor
+     * @param adapter Pointer to iAP2 USB adapter
+     */
+    IAPSignalHandlerDelegate(IAP2USBEmulationTransportAdapter* adapter);
+
+    /**
+     * @brief threadMain Main loop to track incoming signals
+     */
+    void threadMain() OVERRIDE;
+
+    /**
+     * @brief exitThreadMain Stops main loop
+     */
+    void exitThreadMain() OVERRIDE;
+
+    private:
+    IAP2USBEmulationTransportAdapter* adapter_;
+    bool run_flag_;
+    int in_;
+  };
+
+  threads::Thread* signal_handler_;
+  int out_;
```

Comment:
@dev-gh keep in_ and out_ variables near each other

Reply:
@LuxoftAKutsan 
They are maintained by different instances: in - by delegate, out - by adapter. Also pipe opening is blocking operation and happens sequentially so if we'll be managing them with single thread there will be dead lock.